File: app/src/main/java/com/development/allanproject/views/activity/AddExperience.java
package com.development.allanproject.views.activity;

import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.development.allanproject.R;
import com.development.allanproject.databinding.ActivityAddExperienceBinding;

public class AddExperience extends BaseActivity implements View.OnClickListener {
ActivityAddExperienceBinding experienceBinding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        experienceBinding = DataBindingUtil.setContentView(this,R.layout.activity_add_experience);
        experienceBinding.btnNext.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.btn_next:
                startActivity(new Intent(this,AddExperienceInfo.class));
                break;


        }
    }
}

File: app/src/main/java/com/development/allanproject/views/activity/ui/addexperience/AddExperience.kt
package com.development.allanproject.views.activity.ui.addexperience

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.ViewModelProvider
import com.development.allanproject.R
import com.development.allanproject.databinding.ActivityAddExperienceBinding
import com.development.allanproject.model.signupModel.SignResponse
import com.development.allanproject.util.*
import com.development.allanproject.views.activity.ui.personal.PersonDetailViewModel
import com.development.allanproject.views.activity.ui.personal.PersonalDetailViewModelFactory
import kotlinx.android.synthetic.main.activity_personal_detail.*
import kotlinx.android.synthetic.main.activity_sign_up.*
import kotlinx.android.synthetic.main.activity_sign_up.progress_bar
import org.kodein.di.KodeinAware
import org.kodein.di.android.kodein
import org.kodein.di.generic.instance

class AddExperience : AppCompatActivity() , AuthListener, KodeinAware {
    private lateinit var binding: ActivityAddExperienceBinding
    private lateinit var viewModel : AddExperienceViewModel

    override val kodein by kodein()
    private val factory: AddExperienceViewModelFactory by instance()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this,R.layout.activity_add_experience)
        viewModel = ViewModelProvider(this, factory).get(AddExperienceViewModel::class.java)
        binding.viewmodel = viewModel
        viewModel.authListener = this

    }

    override fun onStarted() {
        progress_bar.show()
        toast("Login Started")
    }

    override fun onSuccess(response: SignResponse) {
        progress_bar.hide()
        root_layout.snackbar("${response.step_no} is Done")

    }

    override fun onFailure(message: String) {
        progress_bar.hide()
        root_layout.snackbar(message)
    }
}
